cmd/plugins.go

```diff
@@ -0,0 +1,501 @@
+					if dep.name == pluginName {
+						// plugin in pubspec.lock
+						pluginInUse = true
+						break
+					}
+				}
+
+				if !pluginInUse || cleanPurge {
+					if dryRun {
+						fmt.Printf("       plugin: [%s] can be removed\n", pluginName)
+						continue
+					}
+
+					// remove import file
+					pluginImportPath := filepath.Join(desktopCmdPath, f.Name())
+					err = os.Remove(pluginImportPath)
+					if err != nil {
+						fmt.Printf("hover: Couldn't remove plugin %s: %v\n", pluginName, err)
+						continue
+					}
+					fmt.Printf("       plugin: [%s] removed\n", pluginName)
+
+					// clean-up go.mod
+					pluginImportStr, _ := readPluginGoImport(pluginImportPath, pluginName)
+					// Delete the 'replace' and 'require' import strings from go.mod.
+					// Not mission critical, if the plugins not correctly removed from
+					// the go.mod file, the project still works and the plugin is
+					// successfully removed from the flutter.Application.
+					if pluginImportStr != "" {
+						fileutils.RemoveLinesToFile(filepath.Join(buildPath, "go.mod"), pluginImportStr)
+					}
+				}
+
+			}
+		}
+	},
+}
+
+var pluginGetCmd = &cobra.Command{
+	Use:   "get",
+	Short: "Imports missing platform plugins in the application",
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) > 0 {
+			return errors.New("does not take arguments")
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		assertInFlutterProject()
+		assertHoverInitialized()
+		hoverPluginGet()
+	},
+}
+
+func hoverPluginGetDryRun() bool {
+	dryRun = true
+	defer func() {
+		dryRun = false
+	}()
+	return hoverPluginGet()
+}
+
+func hoverPluginGet() bool {
+	hasNewPlugin := false
+	dependencyList, err := listPlatformPlugin()
+	if err != nil {
+		fmt.Printf("hover: %v\n", err)
+		os.Exit(1)
+	}
+
+	for _, dep := range dependencyList {
+		if !dep.desktop {
+			continue
+		}
+
+		if !dep.autoImport {
+			fmt.Printf("       plugin: [%s] couldn't be imported, check the plugin's README for manual instructions\n", dep.name)
+			continue
+		}
+
+		if dep.imported() {
+			fmt.Printf("       plugin: [%s] already imported\n", dep.name)
+			if !reImport {
+				continue
+			}
+		}
+
+		hasNewPlugin = true
+		if dryRun {
+			fmt.Printf("       plugin: [%s] can be imported\n", dep.name)
+			continue
+		}
+
+		autoImportTemplatePath := filepath.Join(dep.pluginGoSource, "import.go.tmpl")
+		pluginImportOutPath := filepath.Join(buildPath, "cmd", fmt.Sprintf("import-%s-plugin.go", dep.name))
+		if dep.standaloneImpl {
+			fileutils.DownloadFile(dep.pluginGoSource, pluginImportOutPath)
+		} else {
+			fileutils.CopyFile(autoImportTemplatePath, pluginImportOutPath)
+		}
+		fmt.Printf("       plugin: [%s] imported\n", dep.name)
+
+		// if local plugin
+		if dep.path != "" {
+			pluginImportStr, errParse := readPluginGoImport(pluginImportOutPath, dep.name)
+			path, err := filepath.Abs(filepath.Join(dep.path, "go"))
+			if errParse != nil {
+				err = errParse
+			}
```

Comment:
I don't understand? Why not check `err` before the filepath.Abs line?

Reply:
Ok I need some go advice here.

The following warning message needs to be printed if  `readPluginGoImport` or/and `filepath.Abs` are returning an error.

How do I don't duplicate the `fmt.Printf ... warning...` message without introducing a new function/goto..
What is the idiomatic way to handle errors in this context?